### PR TITLE
fix: DBTP-1382 Fix conduit session KMS access permissions

### PR DESCRIPTION
**Pull request by antroy-madetech**
Addresses [DBTP-1382](https://uktrade.atlassian.net/browse/DBTP-1382)

The name of the new role in terraform used by conduit to allow KMS access etc was incorrectly formatted. This PR fixes that and introduces more tests and better error handling around the code getting the role.

---
### Title:
- [x] Scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Ticket reference included (unless it's a quick out of ticket thing)
### Description:
- [x] Link to ticket included (unless it's a quick out of ticket thing)
- [x] Includes tests (or an explanation for why it doesn't)
- [x] If the work includes user interface changes, before and after screenshots included in description
- [x] Includes any applicable changes to the documentation in this code base
- [x] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)
### Tasks:
- [ ] [Trigger the pull request regression tests for this branch](https://github.com/uktrade/platform-tools?tab=readme-ov-file#regression-tests) and confirm that they are passing

[DBTP-1382]: https://uktrade.atlassian.net/browse/DBTP-1382?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dbt_platform_helper/commands/conduit.py [hunk 1]
```diff
@@ -207,15 +209,19 @@ def create_addon_client_task(
...
-        )
-        execution_role = (
-            f"--execution-role {app.name}-{addon_type}-{app.name}-{env}-conduitEcsTask "
-        )
-    except:
+        session.client("iam").get_role(RoleName=role_name)
+        execution_role = f"--execution-role {role_name} "
+    except ClientError as ex:
         execution_role = ""
+        # We cannot catch NoSuchEntityException directly as it is a dynamically generated exception by the boto3
```
Comment (yusufsheiqh) [inline dbt_platform_helper/commands/conduit.py:219]: Would something like `client.exceptions.NoSuchEntityException` not work here?

File: tests/platform_helper/test_command_conduit.py [hunk 2]
```diff
@@ -170,22 +172,64 @@ def test_create_addon_client_task(
...
-    subprocess.call with the correct secret ARN."""
+    subprocess.call with the correct secret ARN and execution role."""
     from dbt_platform_helper.commands.conduit import create_addon_client_task
 
+    env = "development"
+    mock_application = Mock()
+    mock_application.name = "test-application"
+    mock_application.environments = {"development": Mock()}
+
     task_name = mock_task_name(addon_name)
```
Comment (WillGibson) [inline tests/platform_helper/test_command_conduit.py:186]: Have the line break after the arrange lines...
```suggestion
    task_name = mock_task_name(addon_name)
    
```

File: tests/platform_helper/test_command_conduit.py [hunk 3]
```diff
@@ -170,22 +172,64 @@ def test_create_addon_client_task(
...
-    create_addon_client_task(
-        mock_application, "development", addon_type, addon_name, task_name, access
+    create_addon_client_task(mock_application, env, addon_type, addon_name, task_name, access)
+    secret_name = mock_connection_secret_name(mock_application, addon_type, addon_name, access)
```
Comment (WillGibson) [inline tests/platform_helper/test_command_conduit.py:188]: Why are we calling a mock method after the "act"? 

As far as I can tell it's not used.

Question applies to the test below as well.

File: tests/platform_helper/test_command_conduit.py [hunk 4]
```diff
@@ -170,22 +172,64 @@ def test_create_addon_client_task(
...
+    addon_type = "custom-name-postgres"
+    access = "read"
+    env = "development"
+    mock_application.environments[env] = Mock()
+    mock_application.environments[env].session.client.return_value = Mock()
+    mock_application.environments[env].session.client.return_value.get_role.side_effect = (
+        NoSuchEntityException()
     )
+
+    task_name = mock_task_name(addon_name)
```
Comment (WillGibson) [inline tests/platform_helper/test_command_conduit.py:226]:
```suggestion
    task_name = mock_task_name(addon_name)
    
```

File: tests/platform_helper/test_command_conduit.py [hunk 5]
```diff
@@ -195,6 +239,46 @@ def test_create_addon_client_task(
...
+    mock_application.environments[env] = Mock()
+    mock_application.environments[env].session.client.return_value = Mock()
+    mock_application.environments[env].session.client.return_value.get_role.side_effect = (
+        ClientError(
+            operation_name="something_else",
+            error_response={"Error": {"Message": "Something went wrong"}},
+        )
+    )
+
+    task_name = mock_task_name(addon_name)
```
Comment (WillGibson) [inline tests/platform_helper/test_command_conduit.py:269]:
```suggestion
    task_name = mock_task_name(addon_name)
```

File: dbt_platform_helper/commands/conduit.py (hunk 1 above)
Comment (antroy-madetech) [reply dbt_platform_helper/commands/conduit.py:219]: Will check.

File: tests/platform_helper/test_command_conduit.py (hunk 3 above)
Comment (antroy-madetech) [reply tests/platform_helper/test_command_conduit.py]: It was a badly named function. It is an expected name only used in those tests not a mock. Have changed this.

File: .tool-versions [hunk 6]
```diff
@@ -1,2 +1,2 @@
-python 3.12.4 3.11.5 3.10.13 3.9.18
+python 3.11.5
```
Comment (WillGibson) [inline .tool-versions:1]: Why is this change here?

File: dbt_platform_helper/commands/conduit.py [hunk 7]
```diff
@@ -207,15 +209,19 @@ def create_addon_client_task(
...
+        # error factory. Checking the error code is the recommended way of handling these exceptions.
```
Comment (WillGibson) [inline dbt_platform_helper/commands/conduit.py:220]:
```suggestion
        # We cannot check for botocore.errorfactory.NoSuchEntityException as botocore generates that class on the fly as part of errorfactory.
        # Checking the error code is the recommended way of handling these exceptions.
```

File: tests/platform_helper/test_command_environment.py [hunk 8]
```diff
@@ -493,6 +503,7 @@ def test_generate(
         mock_is_terraform_project.return_value = is_terraform
 
         result = CliRunner().invoke(generate, ["--name", "test"])
+        traceback.print_tb(result.exc_info[2])
```
Comment (WillGibson) [inline tests/platform_helper/test_command_environment.py:506]: Temporary debug?

**Review by WillGibson — COMMENTED** [review]
Nearly there

File: .tool-versions (hunk 6 above)
Comment (antroy-madetech) [reply .tool-versions:1]: Mistakenly committed.

File: tests/platform_helper/test_command_environment.py (hunk 8 above)
Comment (antroy-madetech) [reply tests/platform_helper/test_command_environment.py:506]: yep

**Review by WillGibson — APPROVED** [review]
